bpf/process/types/basic.h

```diff
@@ -137,6 +137,30 @@ struct event_config {
  */
 #define MAX_STRING 1024
 
+#ifdef __MULTI_KPROBE
+static inline __attribute__((always_inline)) void
+setup_index(void *ctx, struct msg_generic_kprobe *msg,
+	    struct bpf_map_def *config_map)
+{
+	int idx = (int)get_attach_cookie(ctx);
```

Comment:
If I understand correctly, this is set by the loader and it's different for every link (attachment) of the same program, right?

Reply:
yea, each kprobe has specific index which points to the config_map table
and get_attach_cookie returns that index.. the index is set by config code
before loading